src/main/java/org/jooq/lambda/concurrent/SameExecutorCompletionStage.java

```diff
@@ -28,6 +28,9 @@
 
     @Override
     public final <U> CompletionStage<U> thenApplyAsync(Function<? super T, ? extends U> fn) {
+        if (defaultExecutor == null) {
+            return of(delegate.thenApplyAsync(fn), null);
+        }
```

Comment:
Was this what you had in mind? I think this should work. We only need to use this logic on the methods where we were supplying the default `Executor`